lib/client.js

```diff
@@ -197,10 +197,10 @@ function _connect (client) {
   client[kQueue].pause()
 
   client[kStream].finished(socket, (err) => {
```

Comment:
The actual behavior of how and when this is invoked, with or without error is a slightly dependent on node/readable-stream version. This is a safer way to implement.